pages/blog/_posts/2020-10-01-what-is-short-messages.md

```diff
@@ -0,0 +1,214 @@
+Node 2: "Aah, now I see..."
+```
+
+Or maybe
+
+```
+Node 1: "Hi, I want to talk to you about this in the latest version"
+Node 2: "No"
+Node 1: "Ok, then I want to talk to you about this in version 1 instead"
+Node 2: "Maybe I will talk to you, maybe I will not"
+```
+
+For *TCAP* dialogues there are (mainly) four message types.  `Begin`,
+`Continue`, `End`, `Abort`. Each of the types have an ID (or two, as I
+said, telco is complicated), a component and a dialogue part. The
+component contains the *MAP* messages, and the dialogue part contains
+the version negotiation, but it is only used in the first message from
+both nodes for the version negotiation.
+
+_I think this covers most of it, let's get back to the fun part._
+
+# How does SMS work?
+
+*SMS* was initally implemented because of the wish to send text
+messages to pagers using the phoneline when it was not in use for
+phonecalls. It was decided at a meeting in Oslo to be released to the
+public when some French and German company understood it's
+value. (Don't quote me on any of this).
+
+When you send an *SMS*, a signal goes towards the *SMSC* node in your
+current *SS7* network. This signal is in a form of a packet, and is
+called `MO-Forward-SM`. It stands for "Mobile Originating Forward Short
+Message" meaning it started from your (mobile-)phone.
+
+The *SMSC* then takes your *SMS* and tries to find out where to send
+it. For instance if the recipient's phone is on the same
+operator/network or if it should route it to another operators
+*SMSC*. When it is clear where it should be sent, it goes out as another
+packet, a `MT-Forward-SM`. In this case *MT* stands for Mobile Terminated,
+meaning it goes towards the recipients phone.
+
+The similarities in *MO* and *MT* requests are that they both contain the
+origin and destination addresses as well as the user data (your actual
+text message), and a correlation id which is basically a mapping
+between your sim-card and a temporary id and it is used for privacy
+purposes; you might not want other networks and hackers to know about
+who you are.
+
+For *MT* requests there is also a flag which is called
+`moreMessagesToSend` which is used if (god forbid) you would break the
+protocol and send a text message longer than 115 bytes. The message is
+then breaked up into chunks which starts with an empty *TCAP* `Begin`
+message followed with the actual text message in `Continue`
+messages. In the end (_hehe_) the `End` message is transmitted as a
+response and the transaction is finished.
+
+When one of these chunk (or the full message if lesser than 115 bytes)
+is delivered to the recipient, a response is sent back with the
+`MT-Forward-SM` containing a delivery status.
+
+OR, in the case that you are sending an *SMS* to a recipient in another
+network and the recipient is not available, it might contain the time
+the receiving *SMSC* will hold the message and retry to send it to the
```

Comment:
This is documented as the MSC SM holding area (I think it was called), which may be the visited network MSC handling it, or the recipient's SMSC...